API/projectClass/layersClassRealSLD.m

```diff
@@ -0,0 +1,293 @@
+            % Displays the layers table with numbered rows
+            %
+            % layers.displayLayersTable()
+            array = obj.layersTable;
+            len = size(array,1);
+            if len == 0
+                % Make an empty table for display
+                sz = [1 6];
+                varTypes = {'double','double','double','double','double','double'};
+                varNames = {'Name','Thickness','SLD','Roughness','Hydration','Hydrate with'};
+                dummyTable = table('Size',sz,'VariableTypes',varTypes,'VariableNames',varNames);
+                %dummyTable(:,1) = {[],[],[],[],[],[]};
+                disp(dummyTable);
+                fprintf('\n');
+            end
+            p = 1:len;
+            p = p(:);
+            p = table(p);
+            all = [p array];
+            disp(all);
+            
+        end
+        
+    end
+    
+    methods (Access = protected)
+        
+        function appendNewRow(obj,row)
+            % Appends a row to the layers table. The expected input is
+            % a length six cell array.
+            %
+            % layers.appendNewRow({'New Row','','','','','bulk out'});
+            tab = obj.layersTable;
+            newName = row{1};
+            if any(strcmp(newName,tab{:,1}))
+                error('Duplicate layer names are not allowed');
+            end
+            tab = [tab ; row];
+            obj.layersTable = tab;
+            obj.layersCount = obj.layersCount + 1;
+            obj.layersAutoNameCounter = obj.layersAutoNameCounter + 1;
+        end
+        
+    end
+    
+    methods(Static)
+
+        function row = findRowIndex(name,rowNames)
+            % Find the index of a row in the layers class table given its
+            % name. The expected inputs are the name of the row and the
+            % full list of row names.
+
+            % Strip leading or trailing whitespaces from names
+            rowNames = strip(rowNames);
+            name = strip(name);
+
+            % Compare 'name' to list ignoring case
+            index = strcmpi(name, rowNames);
+            if any(index)
+                % Non-zero value in array is the row index
+                row = find(index);
+            else
+                error('Layer name not found');
+            end
+        end
+
+        function param = findParameter(inputVal, paramNames)
+            % Find whether or not a proposed layer parameter is included
+            % in a list of parameters, or obtain a parameter by index.
+            % The expected inputs are the potential layer parameter value
+            % (either name or index) and a list of parameter names.
+
+            if ischar(inputVal)
+                if ~any(strcmpi(inputVal, paramNames))
+                    error('Parameter %s not recognized', inputVal);
+                end
+                param = inputVal;
+
+            elseif isnumeric(inputVal)
+                paramIndex = floor(inputVal);
```

Comment:
I missed this in the multitype table: I don't think it is a good idea to mask a bad index like this. A user indexing with 5.5 could mean to index with 5 or 6 but this assumes we know what the user wants. Also the user might be calling the wrong function and will not know because we correct their input. In my opinion, it is a well defined behaviour for a non-integer index to crash